spec/models/ability_spec.rb

```diff
@@ -10,4 +10,65 @@
       expect(subject.can?(:manage, :all)).to be(true)
     end
   end
+
+  describe 'region admins' do
+    let(:region_admin) { create(:volunteer) }
+    let(:region) { create(:region) }
+
+    before do
+      create(:assignment, :admin, volunteer: region_admin, region: region)
+    end
+
+    subject { Ability.new(region_admin) }
+
+    describe 'regions' do
+      it 'can update their admin regions' do
+        expect(subject.can?(:update, region)).to eq(true)
+      end
+
+      it 'cannot update other regions' do
+        other_region = create(:region)
+        expect(subject.can?(:update, other_region)).to eq(false)
+      end
+    end
+
+    describe 'food types' do
+      let(:food_type) { create(:food_type, region: region) }
```

Comment:
You may be able to use `build` on a lot of these resources.  If actually persisting the thing isn't required - can speed up this test.

Non-blocking